mikrotik/resource_dhcp_server.go

```diff
@@ -0,0 +1,129 @@
+package mikrotik
+
+import (
+	"context"
+
+	"github.com/ddelnano/terraform-provider-mikrotik/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceDhcpServer() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: createDhcpServer,
+		ReadContext:   readDhcpServer,
+		UpdateContext: updateDhcpServer,
+		DeleteContext: deleteDhcpServer,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"add_arp": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"address_pool": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "static-only",
+			},
+			"authoritative": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "yes",
+			},
+			"disabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"interface": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "*0",
```

Comment:
What does this default value correspond to? The example you made and what I see in my home network use friendly names and the docs aren't clear on what is 'valid'.

Reply:
It is not possible to create a DHCP server w/o assigning it to an interface in console, but if you use API:
```go
func TestAddDhcpServerUpdateAndDelete(t *testing.T) {
	c := NewClient(GetConfigFromEnv())

	name := "myserver"
	disabled := true
	dhcpServer, err := c.AddDhcpServer(&DhcpServer{
		Name:     name,
		Disabled: disabled,
	})
	if err != nil {
		t.Fatal(err)
	}
.....
}
```

the list of servers would be:
```shell
[admin@MikroTik] > /ip dhcp-server print
Flags: D - dynamic, X - disabled, I - invalid
 #    NAME                        INTERFACE                       RELAY           ADDRESS-POOL                       LEASE-TIME ADD-ARP
 0  X dhcp1                       *0                                              static-only                        10m
 1  X dhcp2                       *0                                              static-only                        10m
 2  X dhcp3                       *0                                              static-only                        10m
 3  X dhcp4                       *0                                              static-only                        10m
 4  X dhcp-server                 *0                                              static-only                        10m
 5  X myserver                    *0                                              static-only                        10m
[admin@MikroTik] >
```

notice the `myserver` entry.

The question could lead to necessity of `required` flag in struct tags to indicate that certain field cannot be empty:
```go
type DhcpServer struct {
	Id            string `mikrotik:".id"`
.....
	Interface     string `mikrotik:"interface,required"`
.....
}
```

Reply:
ah, I see. Thanks for that explanation.